executor/common_linux.h

```diff
@@ -1485,6 +1485,139 @@ static long syz_io_uring_submit(volatile long a0, volatile long a1, volatile lon
 
 #endif
 
+#if SYZ_EXECUTOR || __NR_syz_btf_id_by_name
+
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/btf.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define VMLINUX_MAX_SUPPORT_SIZE (10 * 1024 * 1024)
+
+// Read out all the content of /sys/kernel/btf/vmlinux to the buffer and return
+// it.
+static char* read_btf_vmlinux()
+{
+	int fd;
+	struct stat stat;
+	unsigned long vmlinux_size;
+	unsigned long bytes_read = 0;
+
+	static char buf[VMLINUX_MAX_SUPPORT_SIZE];
```

Comment:
Current I store the content of `vmlinux` in this static array because I can't use malloc in the executor code. However, this approach could use a lot of memory and is prone to race condition. What is the right way to do this?

Reply:
One way might be pre-compute once and store the result. However given that we can't use `malloc` I am not sure how to store those strings. I also don't know how to run an function once (I think I will need to deal with race condition here?)